File: Front/Midgar-App/src/app/models/Speaker.ts
import { Events } from "./Events";
import { SocialMedia } from "./SocialMedia";
import { SpeakerEvent } from "./SpeakerEvent";

export interface Speaker {
    id: number;

    name: string;

    speakerInformation: string;

    imageURL: string;

    phone: string;

    email: string;

    socialMedias?: SocialMedia[];
    
    speakerEvents?: Events[];
}

File: Back/src/Midgar.Domain/Entities/Speaker.cs
namespace Midgar.Domain.Entities
{
    public class Speaker
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string SpeakerInformation { get; set; }

        public string ImageURL { get; set; }

        public string Phone { get; set; }

        public string Email { get; set; }

        public IEnumerable<SocialMedia> SocialMedias { get; set; }

        public IEnumerable<SpeakerEvent> SpeakerEvents { get; set; }
    }
}